### DISCUSSION TITLE
Does mediamtx host a server which listens for an rtsp stream?

### DISCUSSION THREAD

**Discussion by kevinjacb** [bluenviron/mediamtx#2226] [Questions] [answered]
I want to host a mediamtx server on an rpi in the same network which listens for an rtsp stream on a port. But with the following configuration:

paths:
  cam:
    source: rtsp://192.168.0.123:11112/mstream

i get the error:

2023/08/21 18:41:58 INF MediaMTX v1.0.0
2023/08/21 18:41:58 INF [path cam] [RTSP source] started
2023/08/21 18:41:58 INF [RTSP] listener opened on :8554 (TCP), :8000 (UDP/RTP), :8001 (UDP/RTCP)
2023/08/21 18:41:58 INF [RTMP] listener opened on :1935
2023/08/21 18:41:58 INF [HLS] listener opened on :8888
2023/08/21 18:41:58 INF [WebRTC] listener opened on :8889 (HTTP)
2023/08/21 18:41:58 INF [SRT] listener opened on :8890 (UDP)
2023/08/21 18:41:58 INF [path cam] [RTSP source] ERR: dial tcp [::1]:11112: connect: connection refused
2023/08/21 18:42:03 INF [path cam] [RTSP source] ERR: dial tcp [::1]:11112: connect: connection refused

I publish stream using ffmpeg: ffmpeg -i input.mp4 -rtsp_transport tcp -vcodec copy -an -f rtsp rtsp://192.168.0.123:11112/mstream

can mediamtx only read from a url? Thank you.

**Comment by aler9** [comment, marked as answer]
Hello, FFmpeg is not a RTSP server, but a RTSP client.

Don't touch the server configuration, leave the default one.

In order to publish a stream from FFmpeg to the server, the command is

```
ffmpeg -re -i input.mp4 -vcodec copy -an -f rtsp rtsp://192.168.0.123:8554/mstream
```

The stream can then be read from another client:

```
ffmpeg -i rtsp://192.168.0.123:8554/mstream ....
```

**Reply by kevinjacb** [reply]
Thank you so much for the clarification! I just understood how it all works and it does so perfectly.

[a comment by an automated account is not shown]
